internal/dirs/copy.go

```diff
@@ -18,5 +22,49 @@ func OverlayDir(dir string) error {
 
 	log.Println("Copying Overlay directory's files")
 
-	return cp.Copy(Overlay, dir)
+	return cp.Copy(Overlay, dir, cp.Options{
+		Sync: true,
+		PreserveOwner: true,
+		PreserveTimes: true,
+		Skip: func(info os.FileInfo, src, dest string) (bool, error) {
+			if info.IsDir() {
+				return false, nil
+			}
+
+			if info.Mode()&os.ModeSymlink != 0 {
+				destPath, err := filepath.EvalSymlinks(dest)
+				if err != nil {
+					return false, err
+				}
+
+				srcPath, err := filepath.EvalSymlinks(src)
+				if err != nil {
+					return false, err
+				}
+
+				if destPath == srcPath {
+					return true, nil
+				}
+			}
+
+			destContent, err := os.ReadFile(dest)
+			if err != nil {
+				return false, nil
+			}
+
+			srcContent, err := os.ReadFile(src)
+			if err != nil {
+				return false, nil
+			}
+
+			if !reflect.DeepEqual(srcContent, destContent) {
+				err = os.Remove(dest)
+				if err != nil {
+					return false, err
+				}
+			}
```

Comment:
This can get incredibly slow, as it uses `reflect`. As this is checking for simple files, it would be a better idea to use `os.SameFile`.

You can use `bytes.Equal` otherwise to compare file contents.

Reply:
Updated!